### ISSUE TITLE
SCOTCH/6.0.6-intel-2018a fails during sanity check

### ISSUE THREAD

**Issue by moravveji** [easybuilders/easybuild-easyconfigs#8509]
Hey guys,

Trying to build `SCOTCH/6.0.6-intel-2018a` I get an error during the sanity check. I have attached EB log (with `--debug` on). 
```bash
== FAILED: Installation ended unsuccessfully (build directory: /mySHM/easybuild/SCOTCH/6.0.6/intel-2018a): build failed (first 300 chars): Sanity check failed: sanity check command acpl -V 2>&1 | grep 'version 6.0.6' exited with code 1 (output: )
```

I have to mention that this SCOTCH is a prerequisite for PETSc, and this ticket is related to the other issue here: https://github.com/easybuilders/easybuild-easyconfigs/issues/8455.

Thanks
Ehsan
[easybuild-quckQt.zip](https://github.com/easybuilders/easybuild-easyconfigs/files/3324474/easybuild-quckQt.zip)

**Comment by wpoely86** [comment]
Can you check the binary of `acpl` in `/myApps/software/SCOTCH/6.0.6-intel-2018a/bin/` ? Does it work if you run it manually?

**Comment by moravveji** [comment]
yes yes yes:
```bash
$> /myApps/software/SCOTCH/6.0.6-intel-2018a/bin/acpl -V
acpl, version 6.0.4
Copyright 2004,2007,2008,2010-2012,2014 IPB, Universite de Bordeaux, INRIA & CNRS, France
This software is libre/free software under CeCILL-C -- see the user's manual for more information
```

**Comment by moravveji** [comment]
I also see a version confusion, in the temporary build directory; check out this path please:
`/mySHM/easybuild/SCOTCH/6.0.6/intel-2018a/scotch_6.0.4/` 
whereas I was expecting to see something like: 
`/mySHM/easybuild/SCOTCH/6.0.6/intel-2018a/scotch_6.0.6/` 
This is a very naive expectation, and I might be wrong for any reason. But, thought of sharing.

**Comment by boegel** [comment]
@moravveji The output shows why the sanity check failed: you're installing SCOTCH 6.0.4 rather than 6.0.6?

My guess is that you're using a SCOTCH 6.0.6 easyconfig which has a download URL for a 6.0.4 source tarball. But that also means you're deliberately ignoring failing checksums (which are there for a good reason), is that so?

See also the changes made in https://github.com/easybuilders/easybuild-easyconfigs/pull/7159/files

**Comment by moravveji** [comment]
In fact, not!
I am really building v. 6.0.6. To be clear, I am attaching the EB file that I use (which is no different from what I would pull from the EB easyconfig repository).
[SCOTCH-6.0.6-intel-2018a.eb.zip](https://github.com/easybuilders/easybuild-easyconfigs/files/3324601/SCOTCH-6.0.6-intel-2018a.eb.zip)

**Comment by moravveji** [comment]
Alright, I adapted the `source_urls` and `checksums`, and tried again. This time it fails with the checksum test:
```bash
== FAILED: Installation ended unsuccessfully (build directory: /mySHM/easybuild/SCOTCH/6.0.6/intel-2018a): build failed (first 300 chars): Checksum verification for /mySources/s/SCOTCH/scotch_6.0.6.tar.gz using 686f0cad88d033fe71c8b781735ff742b73a1d82a65b8b1586526d69729ac4cf failed.
```

If I comment out the `checksums`, EB continues until it fails at the same sanity check step, as before.

**Comment by boegel** [comment]
@moravveji `686f0cad88d033fe71c8b781735ff742b73a1d82a65b8b1586526d69729ac4cf` is the correct SHA256 checksum for `scotch_6.0.6.tar.gz`.

If you have something different, then either the downloaded source tarball is corrupt, or you actually downloaded the source tarball for SCOTCH 6.0.4 but named it `scotch_6.0.6.tar.gz` (I'm not claiming you did this manually, that's exactly what we fixed in #7159).

So please try re-downloading `scotch_6.0.6.tar.gz`, either by removing it and running `eb` again, or by using `eb --force-download ...`

**Comment by moravveji** [comment]
@boegel: Thanks Kenneth for the tip; the `--force-download` worked flawlessly. Closing this ticket now.
